### ISSUE TITLE
haxe.Timer.delay with haxe.MainLoop

### ISSUE THREAD

**Issue by Aurel300** [HaxeFoundation/hxnodejs#115]
```haxe
class Main {
  public static function main():Void {
    haxe.MainLoop.addThread(() -> trace("this prints"));
    haxe.Timer.delay(() -> {
        haxe.MainLoop.addThread(() -> trace("this does not print"));
      }, 0);
  }
}
```

Expected both to print, but only the first `trace` works. This is not a problem with logging – I verified by setting a value on the global object like so:

```haxe
class Main {
  public static function main():Void {
    js.Syntax.code("global.someTestValue = 0");
    haxe.MainLoop.addThread(() -> js.Syntax.code("global.someTestValue = 1"));
    haxe.Timer.delay(() -> {
        haxe.MainLoop.addThread(() -> js.Syntax.code("global.someTestValue = 2"));
      }, 0);
    haxe.Timer.delay(() -> trace(js.Syntax.code("global.someTestValue")), 100);
  }
}
```

(Outputs `1`.)

Compiled with `haxe -lib hxnodejs -main Main -js test.js` and executed with `node test.js`. Node version 8.5.0.

**Comment by nadako** [comment]
Hm, I don't think we have anything specific to haxe.Timer and/or haxe.MainLoop in hxnodejs, meaning that the problem is actually in Haxe. I can look into this, but I think it's @ncannasse area of competence.

**Comment by Aurel300** [comment]
Unfortunately the first snippet in my post (when compiled as `haxe -main Main -js test.js`, then included in an HTML file) works fine in Chrome, i.e. both `trace` commands are executed. So it must be related to Node one way or another. After some further examination, the key difference between the JS generated with `-lib hxnodejs` is:

```js
// without -lib hxnodejs
haxe_EntryPoint.run = function() {
	var nextTick = haxe_EntryPoint.processEvents();
	var $window = window;
	var rqf = $window.requestAnimationFrame || $window.webkitRequestAnimationFrame || $window.mozRequestAnimationFrame;
	rqf(haxe_EntryPoint.run);
};

// with -lib hxnodejs
haxe_EntryPoint.run = function() {
	var nextTick = haxe_EntryPoint.processEvents();
	if(nextTick < 0) {
		return;
	}
	setTimeout(haxe_EntryPoint.run,nextTick);
};
```

The former doesn't work when executed with `node`, since `window` is not defined.

Also (since there is a check for `nextTick`), setting the delay for the second `trace` to a higher number doesn't change the result, although Node does wait for the delayed function to actually execute.

**Comment by nadako** [comment]
Yeah, there's a check here [in the Haxe sources]( https://github.com/HaxeFoundation/haxe/blob/2627715a3531e1fb5a37c8892de74dee3b9ec08a/std/haxe/EntryPoint.hx#L119-L121), I guess something is wrong there...

**Comment by Aurel300** [comment]
Ah yes, I didn't realise Haxe JS produces different code with `-D nodejs`. Should I close this and open an issue in Haxe then?

**Comment by nadako** [comment]
Let's see if GitHub's fancy "transfer issue" feature works :)
